core/src/commonMain/kotlin/entity/Asset.kt

```diff
@@ -0,0 +1,77 @@
+package dev.kord.core.entity
+
+import dev.kord.common.entity.Snowflake
+import dev.kord.core.Kord
+import dev.kord.core.KordObject
+import dev.kord.rest.Image
+import dev.kord.rest.Image.Format.*
+import dev.kord.rest.route.CdnUrl
+import dev.kord.rest.route.DiscordCdn
+
+public class Asset private constructor(
+    public val isAnimated: Boolean,
+    public val cdnUrl: CdnUrl,
+    override val kord: Kord,
+    private val forcedFormat: Image.Format? = null,
+) : KordObject {
+
+    public suspend fun getImage(format: Image.Format? = null, size: Image.Size? = null): Image = Image.fromUrl(
+        client = kord.resources.httpClient,
+        url = cdnUrl.toUrl {
+            this.format = forcedFormat ?: format ?: if (isAnimated) GIF else WEBP
```

Comment:
so webp is the right default format for non-animated assets? just curious because i myself don't know what it should be tbh

Reply:
I don't know either xd

Reply:
@HopeBaron, what do you think?

Reply:
Based on this: https://discord.com/developers/docs/reference#image-formatting-cdn-endpoints
it seems that the default is PNG

Reply:
would also say png seems more reasonable, should be changed then

Reply:
i changed the default to png in https://github.com/kordlib/kord/pull/703/commits/b61555d52fc15519b6aee56bdfb4b0f324aa04e1